github/Requester.py

```diff
@@ -357,14 +370,17 @@ def __init__(
         self.__pool_size = pool_size
         self.__seconds_between_requests = seconds_between_requests
         self.__seconds_between_writes = seconds_between_writes
-        self.__last_requests = defaultdict(lambda: 0.0)
+        self.__last_requests: Dict[str, float] = dict()
         self.__scheme = o.scheme
         if o.scheme == "https":
             self.__connectionClass = self.__httpsConnectionClass
         elif o.scheme == "http":
             self.__connectionClass = self.__httpConnectionClass
         else:
             assert False, "Unknown URL scheme"
+        self.__connection: Optional[Union[HTTPRequestsConnectionClass, HTTPSRequestsConnectionClass]] = None
+        self.__connection_lock = threading.Lock()
+        self.__custom_connections: Deque[Union[HTTPRequestsConnectionClass, HTTPSRequestsConnectionClass]] = deque()
```

Comment:
Do you see problems that we pickle these connection classes including their sessions?

Reply:
`__connection` was pickled previously without issues, I tested pickle `__custom_connections` by append `self.__connection` to it manually, `pickle.loads` and `pickle.dumps` are all good.

Reply:
I am more thinking about what does it mean to pickle these connection instances and reuse them remotely.

Reply:
right, if reuse them remotely, then those connections can't be reused. If it's used for save and restore locally, then maybe those connections are still valid. To support both use cases, I think it's ok to ignore them during pickle. I'll push a new change soon.

Reply:
Ignored in https://github.com/PyGithub/PyGithub/pull/2724/commits/7e17681d87775ecd4b38760487f646ec928eef5f, is there a good test case I can use for testing `__custom_connections`?

Reply:
Found an issue with previous test, use `unittest.TestCase` won't use replay data, use `Framework.TestCase` will cause pickle error because of lambda in test [setup](https://github.com/PyGithub/PyGithub/blob/48d2009c8141e3e32def3db35ecf7d91ce1e687a/tests/Framework.py#L280).
So to avoid external requests in test, I switched to use `lazy=True` in pickle tests.
see https://github.com/PyGithub/PyGithub/pull/2724/commits/49d27405f3e65c877d8d5a9a9ed13d1c203c66e5

Reply:
The `testPickleRepository` test works just fine for me without `lazy=True`.

Reply:
if you turn off your network, that test will fail as it will send real http request to github

Reply:
Another way to verify is to update the replay data to return `404`, the test can still pass without `lazy=True`, because the file is not being used.
Since we don't expect unitests to send real http requests out, and `Framework.TestCase` is not pickable, I updated to be `lazy=True` and removed replay data.

Reply:
Right, testing with `lazy=True` is not ideal, but I agree, test setup is not made for this.